editing/link-boundaries-insertion.html

```diff
@@ -0,0 +1,39 @@
+<!DOCTYPE HTML>
+<meta charset=utf-8>
+<title>Placing selection and typing inside empty elements</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+
+<div contenteditable>
+  <p><a href="https://example.com" id="test-end">Link</a></p>
+  <p><a href="https://example.com" id="test-beginning">Link</a></p>
+</div>
+<script>
+  test( () => {
+    const target = document.querySelector( '#test-end' );
+    const range = document.createRange();
+    const selection = getSelection();
+
+    range.selectNodeContents( target );
+    selection.removeAllRanges();
+    selection.addRange( range );
+    selection.collapseToEnd();
+
+    document.execCommand( 'insertText', false, 'a' );
+    assert_equals( target.innerHTML, 'Linka', 'The text was inserted into the link' );
```

Comment:
Oh, which do browsers pass or fail these tests?

Reply:
Firefox passes both tests, Safari and Chrome fail both.

Reply:
Thanks. Especially Chrome "normalizes" `Selection` before handling edit operations/commands. So the result is what I can imagine. (When I make Firefox emulate the behavior, odd behavior caused by the normalization make the code complicated...)